apps/next/src/pages/AccountManagement/components/NavigateBackContext.tsx

```diff
@@ -0,0 +1,49 @@
+import { createContext, FC, ReactNode, use, useMemo } from 'react';
+import { useHistory } from 'react-router-dom';
+
+type UnregisterFn = () => void;
+type NavigateBackContext = {
+  register: (callback: VoidFunction) => UnregisterFn;
+  goBack: VoidFunction;
+};
+
+type Props = {
+  children: (goBack: VoidFunction) => ReactNode;
+};
+
+const NavigateBackContext = createContext<NavigateBackContext | undefined>(
+  undefined,
+);
+
+export const NavigateBackProvider: FC<Props> = ({ children }) => {
+  const { goBack } = useHistory();
+
+  const context = useMemo<NavigateBackContext>(() => {
+    let onGoBack: VoidFunction | undefined;
+    return {
+      register: (goBackFn: VoidFunction) => {
+        onGoBack = goBackFn;
+        return () => {
+          onGoBack = undefined;
+        };
+      },
+      goBack: () => (onGoBack ?? goBack)(),
```

Comment:
🤔 Does this not work?

```suggestion
      goBack: onGoBack ?? goBack,
```

Reply:
Nope. 
The point of this implementation is to lazy evaluate the actual `goBack` function
Since `register` function doesn't trigger re-render, the component which uses `context.goBack` will always use `history.goBack` function, because at the point of rendering there's a big change that `onGoBack` will be undefined